### ISSUE TITLE
Plugin did not respond: `okta_app_signon_policy_rule` on plans after initial plan/apply was successful

### ISSUE THREAD

**Issue by jodoscott** [okta/terraform-provider-okta#1699]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v1.5.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/dns v3.3.2
+ provider registry.terraform.io/hashicorp/vault v3.16.0
+ provider registry.terraform.io/okta/okta v4.3.0
```
### Affected Resource(s)

- okta_app_signon_policy_rule

### Terraform Configuration Files

```hcl
### Resource in module
resource "okta_app_signon_policy_rule" "this" {
  name                        = var.default ? "Catch-all Rule" : var.name
  policy_id                   = var.policy_id
  priority                    = var.priority
  access                      = var.access
  factor_mode                 = var.factor_mode
  inactivity_period           = var.inactivity_period
  re_authentication_frequency = var.re_authentication_frequency
  type                        = var.type
  network_connection          = var.default ? null : var.network_connection
  device_is_registered        = var.device_is_registered
  device_is_managed           = var.device_is_managed
  device_assurances_included  = var.device_assurances_included

  # Lists of Groups
  groups_excluded = var.default ? null : var.groups_excluded
  groups_included = var.default ? null : var.groups_included

  # List of Users
  users_excluded = var.default ? null : var.users_excluded
  users_included = var.default ? null : var.users_included

  # List of User Types
  user_types_excluded = var.default ? null : var.user_types_excluded
  user_types_included = var.default ? null : var.user_types_included

  # Lists of Networks
  network_includes = var.default ? null : var.network_includes
  network_excludes = var.default ? null : var.network_excludes

  # We must encode each constraint as JSON as well as remove any keys that have null values
  constraints = var.constraints == null ? [] : [

    # Encode each constraint as JSON
    for constraint in var.constraints : jsonencode({

      # Out of the two keys (possession and knowledge), only provide the ones that are not null
      for key, item in constraint : key => {

        # Out of the 7 keys (types, methods, hardwareProtection, deviceBound, phishingResistant, userPresence, reauthenticateIn), only provide the ones that are not null
        for k, v in item : k => v if v != null

      } if item != null
    })
  ]

  dynamic "platform_include" {
    for_each = var.platform_include == null ? [] : var.platform_include
    content {
      type    = platform_include.value.type
      os_type = platform_include.value.os.type
    }
  }
}
```

### Panic Output

https://gist.github.com/jodoscott/3df8ef24b9bb2ceec7b6d403146fc73e

### Actual Behavior
```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│   with module.okta-app-signon-policy-rule["sample-catch-all"].okta_app_signon_policy_rule.this,
│   on modules/okta-app-signon-policy-rule/main.tf line 3, in resource "okta_app_signon_policy_rule" "this":
│    3: resource "okta_app_signon_policy_rule" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
```

### Steps to Reproduce

1. Perform initial resource configuration and plan/apply.
2. Perform a plan against the configuration after the initial apply.

**Comment by monde** [comment]
@jodoscott I see in your gist there is a nil pointer panic in okta/resource_okta_app_signon_policy_rule.go which is v4.3.0 of the provider. I addressed this panic with a nil guard in v4.4.0 https://github.com/okta/terraform-provider-okta/releases can you upgrade your provider and let me know if your issue has been addressed in the latest v4.4.x release?

**Comment by jodoscott** [comment]
@monde Thanks for this, I was able to successfully plan/apply after the upgrade to 4.4.0.
